src/OpenSpaceToolkit/Astrodynamics/Trajectory/State.cpp

```diff
@@ -47,38 +68,12 @@ State State::operator+(const State& aState) const
         throw ostk::core::error::runtime::Wrong("Instant");
     }
 
-    if (this->accessPosition().accessFrame() != aState.accessPosition().accessFrame())
+    if (this->frameSPtr_ != aState.frameSPtr_)
     {
-        throw ostk::core::error::runtime::Wrong("Position Frame");
+        throw ostk::core::error::runtime::Wrong("Frame");
     }
 
-    if (this->accessVelocity().accessFrame() != aState.accessVelocity().accessFrame())
-    {
-        throw ostk::core::error::runtime::Wrong("Velocity Frame");
-    }
-
-    if (this->accessPosition().getUnit() != aState.accessPosition().getUnit())
-    {
-        throw ostk::core::error::runtime::Wrong("Position Unit");
-    }
-
-    if (this->accessVelocity().getUnit() != aState.accessVelocity().getUnit())
-    {
-        throw ostk::core::error::runtime::Wrong("Velocity Unit");
-    }
-
-    return {
-        this->instant_,
-        Position(
-            this->getPosition().getCoordinates() + aState.getPosition().getCoordinates(),
-            this->accessPosition().getUnit(),
-            this->accessPosition().accessFrame()
-        ),
-        Velocity(
-            this->getVelocity().getCoordinates() + aState.getVelocity().getCoordinates(),
-            this->accessVelocity().getUnit(),
-            this->accessVelocity().accessFrame()
-        )};
+    return {this->instant_, this->coordinates_ + aState.coordinates_, this->frameSPtr_};
```

Comment:
yes, it's a good point
actually I was questioning whether these operator functions are used and whether they should be kept now that the state is becoming generalised

even before getting to quaternions, what if your pos and vel are parameterised using spherical (not Cartesian) coordinates? it gets messy. I mean it can be done shifting the responsability to the subset itself (i.e. a coordinate subset should now how to +/- to another subset of it's kind)

(I will now go to PR2 and rename subsets `Position` and `Velocity` to `PositionCartesian` and `VelocityCartesian` so that it is more clear and future proof)

(and maybe I update PR2 to loop over subsets and +/- them individually, not assuming just a +/- algebraic operation for all)

Reply:
added as an open question -> https://loftorbital.atlassian.net/wiki/spaces/FD/pages/2181335613/FDEP-003+Extended+State#Open-Questions